harvester_e2e_tests/integration/test_backup_restore.py

```diff
@@ -0,0 +1,803 @@
+
+        time.sleep(3)
+        response = api_client._get(vmi_path)
+        code = response.status_code
+        vmi = response.json()
+    assert datetime.now() < end_time, (
+        f'Timed-out getting VMI {name}.')
+    return vmi
+
+
+def _wait_for_vm_ip(endpoint, api_client, vm_name, wait_timeout):
+    vmi = _get_vmi(endpoint, api_client, vm_name, wait_timeout)
+
+    end_time = datetime.now() + timedelta(seconds=wait_timeout)
+    ip = _find_ip_address_in_vmi(vmi)
+    while datetime.now() < end_time:
+        if ip != '':
+            break
+
+        time.sleep(3)
+        vmi = _get_vmi(endpoint, api_client, vm_name, wait_timeout)
+        ip = _find_ip_address_in_vmi(vmi)
+
+    assert datetime.now() < end_time, (
+        'Timed-out waiting for VM IP address.')
+    return ip
+
+
+@pytest.fixture(scope='module')
+def backup_target_nfs(request, api_client):
+    nfs_endpoint = request.config.getoption('--nfs-endpoint')
+
+    spec = api_client.settings.BackupTargetSpec.NFS(nfs_endpoint)
+    code, data = api_client.settings.update('backup-target', spec)
+    assert 200 == code, (
+        f'Failed to update backup target to NFS: {data}')
+
+
+@pytest.fixture(scope='module')
+def backup_target_s3(request, api_client):
+    s3_endpoint = request.config.getoption('--s3-endpoint')
+    bucket = request.config.getoption('--bucketName')
+    region = request.config.getoption('--region')
+    access_key_id = request.config.getoption('--accessKeyId')
+    secret_access_key = request.config.getoption('--secretAccessKey')
+
+    spec = api_client.settings.BackupTargetSpec.S3(bucket,
+                                                   region,
+                                                   access_key_id,
+                                                   secret_access_key,
+                                                   s3_endpoint)
+    code, data = api_client.settings.update('backup-target', spec)
+    assert 200 == code, (
+        f'Failed to update backup target to S3: {data}')
+
+
+@pytest.fixture(scope='class')
+def vm_name(unique_name):
+    return f'vm-{unique_name}'
+
+
+@pytest.fixture(scope='class')
+def image_display_name(unique_name):
+    return f'image-{unique_name}'
+
+
+@pytest.fixture(scope='class')
+def restored_vm_name(unique_name):
+    return f'restored-{unique_name}'
+
+
+@pytest.fixture(scope='class')
+def backup_name(unique_name):
+    return f'backup-{unique_name}'
+
+
+@pytest.fixture(scope='class')
+def image(wait_timeout, api_client, image_display_name, need_cleanup):
+    code, data = api_client.images.get(image_display_name)
+    if 200 == code:
+        yield data['metadata']['name']
+    else:
+        code, image = api_client.images.create_by_url(image_display_name, IMAGE_URL)
+        assert 201 == code, (
+            f'Failed to create image: {image}')
+        image_name = image['metadata']['name']
+        _wait_for_image_ready(wait_timeout, api_client, image_name)

```

Comment:
I think we can just write code in this function instead of make another function outside.